lib/internal/Magento/Framework/CompiledInterception/Generator/CompiledPluginList.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\CompiledInterception\Generator;
+
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Interception\PluginList\PluginList;
+use Magento\Framework\Interception\ObjectManager\ConfigInterface;
+use Magento\Framework\ObjectManager\Config\Reader\Dom;
+
+class CompiledPluginList extends PluginList
```

Comment:
This inherits from `PluginList` not to duplicate code. It uses parent class logic to make sure plugins will be executed exactly in the same order and using same logic as they were previously.

Reply:
If you inherit from PluginList then you have to implement anything which is private.  
Properties: $logger, $serializer
Methods: trimInstanceStartingBackslash, filterPlugins, getLogger

Make life easier for anyone who wants to subclass from you and define them as protected.  You are allowed to raise the visibility of a private method in a child since the parent and any siblings don't know about it.